### PR TITLE
repo-updater: Reduce allocations caused by scheduler syncing

### PR DESCRIPTION
**Pull request by ryanslade**
Our syncScheduler loop has two jobs:

1. Ensure all indexable repos are in the scheduler
2. Ensure that any uncloned repos in the scheduler are moved to the
   front of the queue

For 2, we used to fetch ALL cloned repos and send that entire list to
the scheduler which would then compare it's internal state and move any
repo NOT in the cloned list to the front of the queue. This caused a
large number of allocations in repo-updater.

Instead, we now fetch only uncloned repos managed by the scheduler from
the database which will be a much smaller list and can then send this to
the scheduler instead.

### CODE REVIEW

**Comment by ryanslade** [comment]
This change was inspired by looking at allocations on repo-updater and noticing that over 50% of the were caused by listing all repos via gitserver:
https://console.cloud.google.com/profiler/repo-updater;type=HEAP_ALLOC/alloc_space?project=sourcegraph-dev

File: cmd/repo-updater/shared/main.go [hunk 1]
```diff
@@ -406,10 +406,13 @@ type scheduler interface {
 	// UpdateFromDiff updates the scheduled and queued repos from the given sync diff.
 	UpdateFromDiff(repos.Diff)
 
-	// SetCloned ensures uncloned repos are given priority in the scheduler.
-	SetCloned([]string)
+	// SetUncloned ensures uncloned repos are given priority in the scheduler.
+	SetUncloned([]string)
```
Comment (unknwon) [inline cmd/repo-updater/shared/main.go:410]: 😂

**Review by unknwon — APPROVED** [review]

File: cmd/repo-updater/shared/main.go (hunk 1 above)
Comment (ryanslade) [reply cmd/repo-updater/shared/main.go:410]: Should I change something? :)

Comment (ryanslade) [reply cmd/repo-updater/shared/main.go:410]: Maybe, "PrioritiseUncloned"?

**Pull request merged**